changes/2020-07-13_caching-cmm-max-plaintext-length/change.md

```diff
@@ -0,0 +1,153 @@
+[//]: # "SPDX-License-Identifier: CC-BY-SA-4.0"
+
+# Enforce Safe Handling of Max Plaintext Length in Caching Cryptographic Materials Manager
+
+## Affected Features
+
+| Feature                                                                   |
+| ------------------------------------------------------------------------- |
+| [Caching Cryptographic Materials Manager](../../framework/caching-cmm.md) |
+
+## Affected Specifications
+
+| Specification                                                             |
+| ------------------------------------------------------------------------- |
+| [Caching Cryptographic Materials Manager](../../framework/caching-cmm.md) |
+
+## Affected Implementations
+
+| Language   | Repository                                                                            |
+| ---------- | ------------------------------------------------------------------------------------- |
+| C          | [aws-encryption-sdk-c](https://github.com/aws/aws-encryption-sdk-c)                   |
+| Java       | [aws-encryption-sdk-java](https://github.com/aws/aws-encryption-sdk-java)             |
+| JavaScript | [aws-encryption-sdk-javascript](https://github.com/aws/aws-encryption-sdk-javascript) |
+| Python     | [aws-encryption-sdk-python](https://github.com/aws/aws-encryption-sdk-python)         |
+
+## Definitions
+
+### Conventions used in this document
+
+The key words
+"MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT",
+"SHOULD", "SHOULD NOT", "RECOMMENDED", "MAY", and "OPTIONAL"
+in this document are to be interpreted as described in
+[RFC 2119](https://tools.ietf.org/html/rfc2119).
+
+## Summary
+
+The Get Encryption Materials operation accepts an optional `max plaintext length` parameter,
+but the specification does not state how the caching CMM should behave
+when the caller does not provide the parameter value.
+This change specifies that the caching CMM should bypass the cache in this case.
+
+Also, when the caching CMM is performing a Get Encryption Materials operation
+for which no materials are cached,
+it MUST call its underlying CMM's Get Encryption Materials operation.
+The specification does not state what value of `max plaintext length` (if any)
+the caching CMM should pass to its underlying CMM.
+This change specifies that the caching CMM should pass its `byte limit` value
+as the `max plaintext length` parameter of the call to the underlying CMM.
+
+## Out of Scope
+
+- Changing the shape of any CMM APIs is out of scope.
+
+## Motivation
+
+### Fulfulling Encryption Materials Requests Without `max plaintext length`
+
+The Get Encryption Materials operation of the Cryptographic Materials Manager (CMM)
+accepts an Encryption Materials Request that MAY specify a `max plaintext length` field,
+indicating the maximum length of plaintext
+that the caller intends to encrypt using the returned materials.
+The caching Cryptographic Materials Manager (caching CMM)
```

Comment:
np: You should probably move `Cryptographic Materials Manager (caching CMM)` higher up (to the first time you use `caching CMM`) instead of defining it after it's been used a few times.

Reply:
Thanks, I'll fix this.